src/addr_validate.rs

```diff
@@ -0,0 +1,122 @@
+use std::{cell::RefCell, mem::size_of};
+
+use nix::{
+    errno::Errno,
+    unistd::{close, read, write},
+};
+
+thread_local! {
+    static MEM_VALIDATE_PIPE: RefCell<[i32; 2]> = RefCell::new([-1, -1]);
+}
+
+#[inline]
+#[cfg(target_os = "linux")]
+fn create_pipe() -> nix::Result<(i32, i32)> {
+    use nix::fcntl::OFlag;
+    use nix::unistd::pipe2;
+
+    pipe2(OFlag::O_CLOEXEC | OFlag::O_NONBLOCK)
+}
+
+#[inline]
+#[cfg(target_os = "macos")]
```

Comment:
Currently, addr validation is only unabled on Linux. But there is an implementation for macos?

Reply:
Oops. It's a mistake. I'll remove the conditional flag on addr validate.